### PR TITLE
Balder/only use 2 cores for metricsproxy and clustercontroller

### PR DESCRIPTION
**Pull request by baldersheim**
@hmusum PR

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java [hunk 1]
```diff
@@ -110,7 +110,9 @@ public void getConfig(ZookeeperServerConfig.Builder builder) {
 
     @Override
     public void getConfig(QrStartConfig.Builder builder) {
-        builder.jvm(new QrStartConfig.Jvm.Builder().heapsize(512));
+        builder.jvm.heapsize(512);
+        builder.jvm.availableProcessors(2);
+        builder.jvm.verbosegc(false);
```
Comment (hmusum) [inline config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java]: Set heapSizeAsPercentageOfPhysicalMemory to 0 here as done for metrics proxy as well?

Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java]: Correct, but that means that a scenario is not tested. I will add a test for hosted too.

**Review by hmusum — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java [hunk 2]
```diff
@@ -110,7 +110,10 @@ public void getConfig(ZookeeperServerConfig.Builder builder) {
 
     @Override
     public void getConfig(QrStartConfig.Builder builder) {
-        builder.jvm(new QrStartConfig.Jvm.Builder().heapsize(512));
+        builder.jvm.heapsize(512);
+        builder.jvm.heapSizeAsPercentageOfPhysicalMemory(0);
```
Comment (gjoranv) [inline config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java]: It should not be necessary to set the same value as the default in the def file. We should instead avoid setting the wrong values at higher levels in the model. The handling of `getMemoryPercentage` in `ContainerCluster.getConfig(QrStartConfig)` should be moved to the `ApplicationContainerCluster` class. Only those clusters are created by `ContainerModelBuilder`, and that's the only place where we set `memoryPercentage`.

File: config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java [hunk 3]
```diff
@@ -110,7 +110,10 @@ public void getConfig(ZookeeperServerConfig.Builder builder) {
...
+        builder.jvm.availableProcessors(2);
```
Comment (gjoranv) [inline config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java]: Note that the builder setters return the builder itself, so it's not necessary to repeate `builder.` for every setting.

**Review by gjoranv — CHANGES_REQUESTED** [review]

File: config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java (hunk 3 above)
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java]: Fixed

File: config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java (hunk 2 above)
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java]: The intention is not to set it to the same as default config. If you set heapSize you need to ensure that heapSizeAsPercentageOfPhysicalMemory is 0. The fact that it is 0 in default config can change. Now that does not matter.

I was not aware of ApplicationContainerCluster, at least not when memory percentage stuff was written. I will move it there, but as a separate PR.

Comment (gjoranv) [reply config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java]: True. But then we should perhaps set it to 0 in ContainerCluster's getConfig() and override it only in ApplicationContainerCluster.

**Review by gjoranv — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java (hunk 2 above)
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java]: Yes, I will go over and consolidate QrStartConfig generation in the next PR.

**Pull request merged**